save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/s3/DefaultS3Operations.kt

```diff
@@ -34,16 +33,19 @@ class DefaultS3Operations(
 ) : S3Operations, AutoCloseable {
     private val bucketName = properties.bucketName
     private val credentialsProvider: AwsCredentialsProvider = properties.credentials.toAwsCredentialsProvider()
-    private val executorService = with(properties.async) {
```

Comment:
I got very far from this, so a stupid question: if this was an implementation of S3Operations, why you didn't have these `override` methods here before? I do not understand what has changed

Reply:
S3Operations returned `Mono` or `Flux` previously
Now we return `CompletableFuture` to implement implementation of `Storage` using coroutines.
It was done to avoid 2 casting: `CompletableFuture` to `Mono`/`Flux` to `suspend`/`Flow`

Reply:
And where `override` methods appeared from? 
`executorService` - initially you made a private method and now it's an override. Why it wasn't an override from a very beginning? Because of a different type?

Reply:
CompletableFuture is not a right way in case of Spring Flux integration, but I understand your problem with casting. 
Basically that should have been done in Spring S3 integration...

Reply:
make it private